core/domain/email_services.py

```diff
@@ -69,3 +75,165 @@ def get_feedback_thread_reply_info_by_user_and_thread_ids(user_id, thread_id):
     if model is None:
         return None
     return get_feedback_thread_reply_info_from_model(model)
+
+
+def get_incoming_email_address(reply_to_id):
+    """Gets the incoming email address. The client is responsible for recording
+    any audit logs.
+
+    Args:
+        reply_to_id: str. The unique id of the sender.
+
+    Returns:
+        str. The email address of the sender.
+    """
+    return 'reply+%s@%s' % (reply_to_id, feconf.INCOMING_EMAILS_DOMAIN_NAME)
+
+
+def is_email_valid(email_address):
+    """Determines whether an email address is invalid.
+
+    Args:
+        email_address: str. Email address to check.
+
+    Returns:
+        bool. Whether the specified email address is valid.
+    """
+    if not isinstance(email_address, python_utils.BASESTRING):
+        return False
+
+    stripped_address = email_address.strip()
+    if not stripped_address:
+        return False
+    # Regex for a valid email.
+    regex = r'^.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}|[0-9]{1,3})(]?)$'
+    return re.search(regex, email_address)
+
+
+def is_sender_email_valid(sender_email):
+    """Gets the sender_email address and validates that it is of the form
+    'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or 'email_address'.
+
+    Args:
+        sender_email: str. The email address of the sender.
+
+    Returns:
+        bool. Whether the sender_email is valid.
+    """
+    # Checks the case where sender_email is of the form 'Jane <EMAIL>'.
```

Comment:
Why are we checking for the number of names? This doesn't match either the docstring or the previous version of the checks, and I don't know that there is any requirement for the sender's name to have at most two space-separated words in it.

Try not to change the functionality/behaviour from before -- this is why you write tests. (You should probably add some backend tests that would have caught this error.)

Reply:
Done, in actuality the original implementation didn't really check for the whether the email or name existed but since that didn't really make sense with the method name, I added this check. Fixed now.